clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -215,6 +215,12 @@ protected void onJoinComplete(int generation,
 
         // update partition assignment
         subscriptions.assignFromSubscribed(assignment.partitions());
+        this.joinedSubscription = subscriptions.subscription();
```

Comment:
@guozhangwang A metadata fetch could return in the middle of a rebalance (i.e. with the JoinGroup on the wire). In this case, we may end up discovering a new topic that we didn't know about before and adding it to our subscription (see `subscribeFromPattern`). We are able to catch this case currently because although the subscription has changed, `joinedSubscription` has not. With this patch, this is no longer true.